packages/devtools_app/lib/src/inspector/inspector_tree_controller.dart

```diff
@@ -588,98 +596,82 @@ class InspectorTreeController extends Object
 
   /* Search support */
   @override
-  void previousMatch() {
-    super.previousMatch();
-    final currentSearchMatches = searchMatches.value;
-    for (int i = 0; i < currentSearchMatches.length; i++) {
-      final searchMatch = currentSearchMatches[i];
-
-      if (searchMatch.isSelected) {
-        final rowToSelect = i == 0
-            ? currentSearchMatches[currentSearchMatches.length - 1]
-            : currentSearchMatches[i - 1];
-
-        onSelectRow(rowToSelect);
-        break;
-      }
-    }
-  }
-
-  @override
-  void nextMatch() {
-    super.nextMatch();
-    final currentSearchMatches = searchMatches.value;
-    for (int i = 0; i < currentSearchMatches.length; i++) {
-      final searchMatch = currentSearchMatches[i];
-
-      if (searchMatch.isSelected) {
-        final rowToSelect = i == currentSearchMatches.length - 1
-            ? currentSearchMatches[0]
-            : currentSearchMatches[i + 1];
-
-        onSelectRow(rowToSelect);
-        break;
-      }
-    }
+  void onMatchChanged(int index) {
+    onSelectRow(searchMatches.value[index]);
   }
 
   void dispose() {
+    _searchOperation?.cancel();
     if (_searchDebounce?.isActive ?? false) {
       _searchDebounce.cancel();
     }
   }
 
-  Future _searchOperation;
+  CancelableOperation<void> _searchOperation;
   Timer _searchDebounce;
 
+  List<InspectorTreeRow> _searchResults = [];
+
+  @override
+  void resetSearch() {
+    _searchResults = [];
+    super.resetSearch();
+  }
+
+  @override
+  set search(String search) {
+    _searchResults = [];
+    super.search = search;
+
+    _startSearchDebounceTimer();
```

Comment:
This debounceTimer seems like it would be useful for all searching. Instead of overriding the search mixin methods to use it, feel free to add it to the search code directly. to be safe, you could add a parameter that allows each consumer of the search mixin to opt in or out.